--- dwm/config.def.h	2025-08-09 15:00:55.740267680 +0200
+++ dwmmod/config.def.h	2025-08-29 15:13:03.290139971 +0200
@@ -55,6 +55,16 @@
 /* helper for spawning shell commands in the pre dwm-5.0 fashion */
 #define SHCMD(cmd) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } }

+#include <X11/XF86keysym.h>
+static const char *up_vol[]   = { "pactl", "set-sink-volume", "@DEFAULT_SINK@", "+10%",   NULL };
+static const char *down_vol[] = { "pactl", "set-sink-volume", "@DEFAULT_SINK@", "-10%",   NULL };
+static const char *mute_vol[] = { "pactl", "set-sink-mute",   "@DEFAULT_SINK@", "toggle", NULL };
+static const char *mute_mic[] = { "pactl", "set-source-mute",   "@DEFAULT_SOURCE@", "toggle", NULL };
+static const char *brighter[] = { "brightnessctl", "set", "10%+", NULL };
+static const char *dimmer[]   = { "brightnessctl", "set", "10%-", NULL };
+static const char *play[]   = { "playerctl", "play-pause", NULL };
+static const char *stop[]   = { "playerctl", "stop", NULL };
+
 /* commands */
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
@@ -95,6 +105,14 @@
 	TAGKEYS(                        XK_8,                      7)
 	TAGKEYS(                        XK_9,                      8)
 	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
+	{ 0,    XF86XK_AudioMute,                  spawn,          {.v = mute_vol } },
+	{ 0,    XF86XK_AudioLowerVolume,           spawn,          {.v = down_vol } },
+	{ 0,    XF86XK_AudioRaiseVolume,           spawn,          {.v = up_vol } },
+	{ 0,    XF86XK_AudioMicMute,               spawn,          {.v = mute_mic } },
+	{ 0,    XF86XK_MonBrightnessDown,          spawn,          {.v = dimmer } },
+	{ 0,    XF86XK_MonBrightnessUp,            spawn,          {.v = brighter } },
+	{ 0,    XF86XK_AudioPlay,                  spawn,          {.v = play } },
+	{ 0,    XF86XK_AudioStop,                  spawn,          {.v = stop } }
 };

 /* button definitions */
